internal/operators/manager.go

```diff
@@ -1,155 +1,281 @@
+	for opType := range pendingOperators {
+		operator := mgr.operators[opType]
+		result := api.ValidationResult{
+			Status:       api.Success,
+			ValidationId: operator.GetHostValidationID(),
+			Reasons: []string{
+				fmt.Sprintf("%v is disabled", opType),
+			},
+		}
+		results = append(results, result)
+	}
+	return results, nil
 }
 
-// GetOperatorStatus gets status of an operator of given type.
-func (mgr *Manager) GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string {
-	operator, err := findOperator(cluster, operatorType)
+// ValidateCluster validates cluster requirements
+func (mgr *Manager) ValidateCluster(ctx context.Context, cluster *common.Cluster) ([]api.ValidationResult, error) {
+	// TODO: cluster should already contain up-to-date list of operators - implemented here for now to replicate
+	// the original behaviour
+	err := mgr.UpdateDependencies(cluster)
+	if err != nil {
+		return nil, err
+	}
+	clusterOperators, err := unmarshallOperators(cluster.Operators)
 	if err != nil {
-		return "Something went wrong with Unmarshalling operators"
+		return nil, err
 	}
-	if operator != nil {
-		if swag.BoolValue(operator.Enabled) {
-			return operator.Status
+
+	results := make([]api.ValidationResult, 0, len(mgr.operators))
+
+	pendingOperators := make(map[models.OperatorType]bool)
+	for k := range mgr.operators {
+		pendingOperators[k] = true
+	}
+
+	for _, clusterOperator := range clusterOperators {
+		operator := mgr.operators[clusterOperator.OperatorType]
+		var result api.ValidationResult
+		if operator != nil {
+			if swag.BoolValue(clusterOperator.Enabled) {
+				result, err = operator.ValidateCluster(ctx, cluster)
+				if err != nil {
+					return nil, err
+				}
+				delete(pendingOperators, clusterOperator.OperatorType)
+				results = append(results, result)
+			}
+		}
+	}
+	// Add successful validation result for disabled operators
+	for opType := range pendingOperators {
+		operator := mgr.operators[opType]
+		result := api.ValidationResult{
+			Status:       api.Success,
+			ValidationId: operator.GetClusterValidationID(),
+			Reasons: []string{
+				fmt.Sprintf("%v is disabled", opType),
+			},
 		}
+		results = append(results, result)
 	}
-	return "OCS is disabled"
+	return results, nil
 }
 
-func (mgr *Manager) generateLSOManifests(cluster *common.Cluster) (map[string]string, error) {
-	mgr.log.Info("Creating LSO Manifests")
-	return lso.Manifests(cluster.OpenshiftVersion)
-}
+// UpdateDependencies amends the list of requested additional operators with any missing dependencies
+func (mgr *Manager) UpdateDependencies(cluster *common.Cluster) error {
+	operators, err := unmarshallOperators(cluster.Operators)
+	if err != nil {
+		return err
+	}
+	operators, err = mgr.resolveDependencies(operators)
+	if err != nil {
+		return err
+	}
+	updatedOperators, err := json.Marshal(operators)
+	if err != nil {
+		return err
+	}
+	cluster.Operators = string(updatedOperators)
```

Comment:
I'm not sure but maybe we should update the cluster Operators also in DB.

Reply:
I wanted to keep the behavior unchanged compared to the original version; if you think this should be done within this PR, I will add proper changes.